scripts/build/builders/host.py

```diff
@@ -45,21 +46,61 @@ def BinaryName(self):
             raise Exception('Unknown app type: %r' % self)
 
 
-def ConcretePlatformName():
-    uname_result = uname()
-    return '-'.join([uname_result.system.lower(), release(), uname_result.machine])
+class HostBoard(Enum):
+    NATIVE = auto()
+
+    # cross-compile support
+    CROSS_COMPILE_ARM64 = auto()
+
+    def PlatformName(self):
+        if self == HostBoard.NATIVE:
+            uname_result = uname()
+            return '-'.join([uname_result.system.lower(), uname_result.machine])
+        elif self == HostBoard.CROSS_COMPILE_ARM64:
+            return 'linux-rpi'
+        else:
+            raise Exception('Unknown host board type: %r' % self)
 
 
 class HostBuilder(GnBuilder):
 
-    def __init__(self, root, runner, output_prefix: str, app: HostApp):
+    def __init__(self, root, runner, output_prefix: str, app: HostApp, board=HostBoard.NATIVE):
         super(HostBuilder, self).__init__(
             root=os.path.join(root, 'examples', app.ExamplePath()),
             runner=runner,
             output_prefix=output_prefix)
 
         self.app_name = app.BinaryName()
         self.map_name = self.app_name + '.map'
+        self.board = board
+
+    def GnBuildArgs(self):
+        if self.board == HostBoard.NATIVE:
+            return None
+        elif self.board == HostBoard.CROSS_COMPILE_ARM64:
+            return [
+                'target_cpu="arm64"',
+                'is_clang=true',
+                'chip_crypto="mbedtls"',
+                'sysroot="%s"' % self.SysRootPath('SYSROOT_AARCH64'),
+            ]
+        else:
+            raise Exception('Unknown host board type: %r' % self)
+
+    def GnBuildEnv(self):
+        if self.board == HostBoard.NATIVE:
+            return None
+        elif self.board == HostBoard.CROSS_COMPILE_ARM64:
+            return {
+                'PKG_CONFIG_PATH': self.SysRootPath('SYSROOT_AARCH64') + '/lib/aarch64-linux-gnu/pkgconfig',
```

Comment:
Why is this needed?

Reply:
```
2021-09-23 14:47:39 INFO    Package avahi-client was not found in the pkg-config search path.
2021-09-23 14:47:39 INFO    Perhaps you should add the directory containing `avahi-client.pc'
2021-09-23 14:47:39 INFO    to the PKG_CONFIG_PATH environment variable
2021-09-23 14:47:39 INFO    No package 'avahi-client' found
2021-09-23 14:47:39 INFO    Traceback (most recent call last):
2021-09-23 14:47:39 INFO      File "/workspaces/connectedhomeip/examples/all-clusters-app/linux/third_party/connectedhomeip/build/config/linux/pkg-config.py", line 259, in <module>
2021-09-23 14:47:39 INFO        sys.exit(main())
Traceback (most recent call last):
2021-09-23 14:47:39 INFO      File "/workspaces/connectedhomeip/examples/all-clusters-app/linux/third_party/connectedhomeip/build/config/linux/pkg-config.py", line 156, in main
2021-09-23 14:47:39 INFO        prefix = GetPkgConfigPrefixToStrip(options, args)
2021-09-23 14:47:39 INFO      File "/workspaces/connectedhomeip/examples/all-clusters-app/linux/third_party/connectedhomeip/build/config/linux/pkg-config.py", line 100, in GetPkgConfigPrefixToStrip
2021-09-23 14:47:39 INFO        prefix = subprocess.check_output([options.pkg_config,
2021-09-23 14:47:39 INFO      File "/usr/lib/python3.9/subprocess.py", line 424, in check_output
2021-09-23 14:47:39 INFO        return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
2021-09-23 14:47:39 INFO      File "/usr/lib/python3.9/subprocess.py", line 528, in run
2021-09-23 14:47:39 INFO        raise CalledProcessError(retcode, process.args,
2021-09-23 14:47:39 INFO    subprocess.CalledProcessError: Command '['pkg-config', '--variable=prefix', 'avahi-client']' returned non-zero exit status 1.
2021-09-23 14:47:39 INFO    
2021-09-23 14:47:39 INFO    See //third_party/connectedhomeip/src/platform/BUILD.gn:34:3: whence it was called.
2021-09-23 14:47:39 INFO      pkg_config("avahi_client_config") {
2021-09-23 14:47:39 INFO      ^----------------------------------
2021-09-23 14:47:39 INFO    See //third_party/connectedhomeip/src/lib/BUILD.gn:30:5: which caused the file to be included.
2021-09-23 14:47:39 INFO        "${chip_root}/src/platform",
2021-09-23 14:47:39 INFO        ^--------------------------
```

Reply:
Ok, let's figure that out.. The idea with 'sysroot' param is that's all you need, it's broken if you also need an environment variable.